src/plugin/persist.js

```diff
@@ -239,7 +239,10 @@ eg.module("persist", ["jQuery", window, document], function($, global, doc) {
 			key = GLOBAL_KEY;
 			data = arguments.length === 1 ? state : null;
 		}
-		data && setStateByKey(key, data);
+
+		if (data || arguments.length === 2) {
+			setStateByKey(key, data);
+		}
```

Comment:
understood. Not only that line, I tried to point the whole function.
How about this?

```js
$.persist = function(state, data) {
	var key;

	if (typeof state === "string") {
		key = state;
	} else {
		key = GLOBAL_KEY;
		data = arguments.length === 1 ? state : null;
	}

	if (data || arguments.length === 2) {
		setStateByKey(key, data);
		return data;
	} else {
		return getStateByKey(key);
	}
};
```